lib/maths/CBoostedTreeFactory.cc

```diff
@@ -0,0 +1,325 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+#include <maths/CBoostedTreeFactory.h>
+
+#include <maths/CBayesianOptimisation.h>
+#include <maths/CBoostedTreeImpl.h>
+
+namespace ml {
+namespace maths {
+
+CBoostedTreeFactory::TBoostedTreeUPtr CBoostedTreeFactory::build() {
+    this->initializeMissingFeatureMasks(*m_Frame);
```

Comment:
This code SIGSEGVs if the `m_Frame` wasn't set before build was called. As I mentioned elsewhere, I'd prefer to pass by reference to `constructFromParameters` to ensure this can't happen.

Reply:
As described above, now this should be impossible.